rpcs3/rpcs3qt/update_manager.cpp

```diff
@@ -482,16 +484,28 @@ bool update_manager::handle_rpcs3(const QByteArray& data, bool auto_accept)
 	Byte* outBuffer   = nullptr;
 	usz outBufferSize = 0;
 
-	// Creates temp folder for moving active files
+#ifdef _WIN32
+	// Create temp folder for moving active files
 	const std::string tmp_folder = exe_dir + "rpcs3_old/";
+#else
+	// Create temp folder for extracting the new app
+	const std::string tmp_folder = fs::get_temp_dir() + "rpcs3_new/";
+#endif
 	fs::create_dir(tmp_folder);
 
 	for (UInt32 i = 0; i < db.NumFiles; i++)
 	{
 		usz offset           = 0;
 		usz outSizeProcessed = 0;
-		const bool isDir = SzArEx_IsDir(&db, i);
-		const usz len    = SzArEx_GetFileNameUtf16(&db, i, nullptr);
+		const bool isDir     = SzArEx_IsDir(&db, i);
+		const DWORD attribs  = SzBitWithVals_Check(&db.Attribs, i) ? db.Attribs.Vals[i] : 0;
+#ifdef _WIN32
+		const bool isSymlink = (attribs & FILE_ATTRIBUTE_REPARSE_POINT) != 0;
```

Comment:
Why would the zip contain symlinks?

Reply:
The macOS app bundle contains frameworks like Qt. Framework bundles are structured in a way such that they store the actual contents at something like ‘Versions/A/…’, and alongside that they have a symlink called ‘Current’ that points to the current version. Practically the bundled frameworks only contain the one version, but we need to preserve the symlinks pointing into it.

Reply:
And if it’s about the WIN32 path here specifically, I included it for completeness but if you want I can limit the symlink support to just Mac

Reply:
I'd rather have isSymlink set to false on windows until we actually need it.
You can keep the code as comment.